### DISCUSSION TITLE
Adding custom fields to ALL assets

### DISCUSSION THREAD

**Discussion by robgutshall** [snipe/snipe-it#15061] [Q&A]
I'm a new Snipe-IT user and am pitching it to Admin for use here.  I have it set up and working.  We are going to use it for more than just IT stuff. (Think Maintenance Equipment, Furniture...etc)  I've been tasked to add about 15 custom fields and have done so.  Now I'm lost at the "Asset Model" stuff.  I've gone through some youtube vids and they all say the same thing.  Seems like I have to CHOOSE an item in the inventory then I can set up the asset model.   First, I have nothing in the inventory yet.  2nd I'd want the custom fields to show up every time I add an asset no matter the make or model.
I feel like I'm missing something here.  
Any guidance appreciated

**Comment by TShiremanLCOG** [comment]
I don't believe the "I have to CHOOSE an item in the inventory then I can set up the asset model" is quite correct but maybe I misunderstand.

As a quick review for clarity (which you may already know), think of a Model like a template. For example, a "Dell Precision 4321" laptop. You can have many of that same model. That specific model with serial number 12345 is an Asset. The other one with serial number 54321 is a separate Asset. 

When creating an asset there is general information you must add. You must add an Asset Tag (typically a unique value), then choose a Model, and finally set the Status. Once you choose a Model, more fields will appear and then information can be added based on the model. For example, if you have a custom field for "Toner SKU", you probably only want that on Models that related to printers. Or a custom field for "Desk is electric?" should probably only apply to models that are Furniture or Desks. So how do we make that happen?

My recommendation is to start by creating a few Categories. Say Workstations, Furniture, and then Copiers. Now create a few Models. I know there's an option when creating each model to add a Fieldset, but don't do it yet! Maybe add 1-2 in each Category. So add some workstations and laptops to Workstations, some desks and chairs to Furniture, and some printers and copiers to Copiers. Great!

Now we get to the custom fields. There's three very important things here: 1.) You create individual fields to hold the data. 2.) Those fields are added to a Fieldset. 3.) Each model can only have ONE Fieldset.

Let's talk about Fieldsets. With Workstations, maybe you want to know the screen size and number of USB ports. So you would create a Field for Screen Size and one for USB Ports. Then you create a Workstations Fieldset. For Furniture, maybe you want to know the tabletop dimensions and height. So you create a Table Width, Table Depth, and Height Fields and create a Furniture Info Fieldset. I think you get the idea.

Now that you have Fieldsets, let's revisit the Models. Go back to each one and choose the correct Fieldset for each one. You can probably match up with Fieldset goes with which Model.

Originally you said you had a bunch of custom fields that you want for EVERY model. Without understanding more, I'm going to recommend against that. What that would entail is adding those fields to EVERY fieldset. That doesn't sound right. But, if you can explain more maybe there's a use case or a better way to accomplish that.

**Comment by MasterCakeX5** [comment]
I also would like to have more variability in this regard.
There are very basic fields missing like "room number" and it is utopy to expect somebody who enters a new model to think about assigning the fieldset every time. Especially with multiple agents the chaos is pre-programmed with so much freedom.
I would like to have fields that are enforced across everything in snipe like said room number.
Or fieldsets that are assignable to a category, not a model. Every workstation has a MAC, there is no need to assign these fields to a specific model every time.

**Comment by snipe** [comment]
You can easily add a custom field to all models using the Custom Field Edit screen.


https://github.com/user-attachments/assets/bc73e20b-7c77-4d1a-beab-3f61f0f7ac31

**Reply by MasterCakeX5** [reply]
I know that but this is not what I mean at all.

There are people using the system who have no idea and therefore no permissions for these things. They need to add an asset to the asset management, try to find a model and when they dont find one, they will add one on the fly with the "danger" of missing the fieldset. There is no automatic way to "see" this mistake.
Our basic fields should be already there without going into several other menus and assigning stuff that could be - with mishandling - leading to stuff being added where it doesnt belong, deleted and so on.

So I wonder why there is no option to implement an "idiot-resistant" (or also called realistic) method of polulating custom fields throughout the entirity of the system, fields that would never change anyway, like rooms - unrelated to specific models or at least related to specific pre-defined categories like workstations or laptops.

Now imagine a workplace where every day 10 new models come into the system - that workflow wouldnt be applicable there in some realistic manner if every time the asset management admin would have to manually update and check a model list.

Maybe I overlooked a way to add fieldsets to a category - but as far as I saw there is always the models themselves involved, not a category above it.

Wish:
Fieldset "general" -> applied system-wide with a checkbox
or
Fieldset "workstation" -> assigned to "workstation" category (**not** manually to each workstation model afterwards)
What is inside "workstation" category is unimportant because every model inside "workstation" will receive the fieldset through inheritance/being a member or the "workstation" group.
